lib/block-editor.php

```diff
@@ -0,0 +1,293 @@
+<?php
+/**
+ * Block Editor API.
+ *
+ * @package Gutenberg
+ * @subpackage Editor
+ * @since 10.5.0
+ */
+
+/**
+ * Returns the list of default categories for block types.
+ *
+ * This is a temporary solution until the Gutenberg plugin sets
+ * the required WordPress version to 5.8.
+ *
+ * @see https://core.trac.wordpress.org/ticket/52920
+ *
+ * @since 10.5.0.
+ *
+ * @return array[] Array of categories for block types.
+ */
+function gutenberg_get_default_block_categories() {
+	return array(
+		array(
+			'slug'  => 'text',
+			'title' => _x( 'Text', 'block category' ),
+			'icon'  => null,
+		),
+		array(
+			'slug'  => 'media',
+			'title' => _x( 'Media', 'block category' ),
+			'icon'  => null,
+		),
+		array(
+			'slug'  => 'design',
+			'title' => _x( 'Design', 'block category' ),
+			'icon'  => null,
+		),
+		array(
+			'slug'  => 'widgets',
+			'title' => _x( 'Widgets', 'block category' ),
+			'icon'  => null,
+		),
+		array(
+			'slug'  => 'theme',
+			'title' => _x( 'Theme', 'block category' ),
+			'icon'  => null,
+		),
+		array(
+			'slug'  => 'embed',
+			'title' => _x( 'Embeds', 'block category' ),
+			'icon'  => null,
+		),
+		array(
+			'slug'  => 'reusable',
+			'title' => _x( 'Reusable Blocks', 'block category' ),
+			'icon'  => null,
+		),
+	);
+}
+
+/**
+ * Returns all the categories for block types that will be shown in the block editor.
+ *
+ * This is a temporary solution until the Gutenberg plugin sets
+ * the required WordPress version to 5.8.
+ *
+ * @see https://core.trac.wordpress.org/ticket/52920
+ *
+ * @since 10.5.0
+ *
+ * @param string|WP_Post $editor_name_or_post The name of the editor (e.g. 'post-editor')
+ *                                            or the post object.
+ *
+ * @return array[] Array of categories for block types.
+ */
+function gutenberg_get_block_categories( $editor_name_or_post ) {
+	// Assume the post editor when the WP_Post object passed.
+	$editor_name        = is_object( $editor_name_or_post ) ? 'post-editor' : $editor_name_or_post;
+	$default_categories = get_default_block_categories();
```

Comment:
Hey @gziolo 👋 I was trying to use `gutenberg_get_block_editor_settings` in https://github.com/WordPress/gutenberg/pull/29969 and got an error about `get_default_block_categories` being `undefined` so I was wondering if this should be:

```
$default_categories = gutenberg_get_default_block_categories();
```

Instead?

Reply:
You are probably correct. Now, it makes me think why it wasn't caught by unit tests 😄

Reply:
https://github.com/WordPress/gutenberg/blob/fe379134644e6a8b43a220d1d6e956def9b5295e/phpunit/block-editor.php#L76

and

https://github.com/WordPress/gutenberg/blob/fe379134644e6a8b43a220d1d6e956def9b5295e/phpunit/block-editor.php#L99

Maybe those tests don't run?

Reply:
Yes, I missed that the file needs a `-test.php` suffix 😅

Reply:
PR ready: https://github.com/WordPress/gutenberg/pull/30984.